### PR TITLE
Ignore RuntimeError when moving in word.

### PR DESCRIPTION
**Pull request by feerrenrut**
Fix for #7009

When opening a blank word document there seems to be a "virtual line-feed" character. The caret can not be moved past this character (I.E. you can not insert after it). It can be selected and copied, but not deleted. It acts as an exclusive end in `textInfo` end offset. The exception is thrown when the new end offset is less than the old end offset. When calling collapse (with `end=true`) on an object that has its end offset set to this "virtual line-feed" character, the `endOffset` is reduced by one, since start can not be this final character (since its exclusive).

The exception seen in #7009 is used to control the program flow for say-all. Several other options for fixing this have been considered:

### Returning an error code: 
This would require that all implementations of collapse return appropriate values.

### Move the check to the say-all code.
Since say-all seems to be the only thing depending on this functionality (to stop an infinite loop), we could move this check to there. To do this we would be relying on `textInfo.copy()` to cache the pre `collapse` value, and then `textInfo.compareEndpoints()` to see if the `endOffset` was now less. Since there is some concern that `compareEndpoints()` does not behave consistently in all implementations this option is considered too risky.

### Ignore this exception when moving the caret
This PR proposes handling this (by ignoring the exception) in the `_caretMovementScriptHelper` function in `cursorManager.py`.

### CODE REVIEW

File: source/cursorManager.py [hunk 1]
```diff
@@ -115,8 +115,15 @@ def _caretMovementScriptHelper(self,gesture,unit,direction=None,posConstant=text
 		if not self._lastSelectionMovedStart and not oldInfo.isCollapsed:
 			info.move(textInfos.UNIT_CHARACTER,-1)
 		if posUnit is not None:
+			# expand and collapse to ensure that we are aligned with the end of the intended unit
 			info.expand(posUnit)
-			info.collapse(end=posUnitEnd)
+			try:
+				info.collapse(end=posUnitEnd)
+			except RuntimeError:
+				# MS Word has a "virtual linefeed" at the end of the document which can cause RuntimError to be raised.
```
Comment (michaelDCurran) [inline source/cursorManager.py]: runtimError -> RuntimeError

**Review by michaelDCurran — APPROVED** [review]

**Comment by feerrenrut** [comment]
Incubated in [commit 90941f4656](https://github.com/nvaccess/nvda/commit/90941f465610d221862d1c74fa5a119baa1f0973) but the comment refered to the wrong PR. I will apply the labels manually.

**Comment by derekriemer** [comment]
Should we close https://github.com/nvaccess/nvda/issues/7009 now?

**Comment by feerrenrut** [comment]
Opps, I wrote "Fix for..." instead of "Fixes..." I'll close it now.

**Pull request merged**